src/dmd/dclass.d

```diff
@@ -992,6 +992,24 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
     {
         v.visit(this);
     }
+
+	/****************************************
+	 * Gets an array of all parents of this class. This includes all parent
+	 * classes, all interfaces implemented by this class, and all interfaces
+	 * implemented by parent classes.
+	 */
+	final BaseClass*[] allParents()
+	{
+		if (baseClass is null)
+			return [];
+		BaseClass*[] parents = baseClass.allParents();
```

Comment:
As a general rule of thumb, use `Array!(BaseClass*)` instead of built-in arrays.  I think there's already a handy `BaseClasses` alias in `dmd.arraytypes`.

Reply:
You can of course still return a slice to `dmd.root.rmem.xcalloc`'d memory in order to keep the dynamic array return type.  The main red flag is the `~=` operation.

Reply:
I've modified this so it no longer returns a dynamically allocated array. It now takes a `BaseClasses` as an input, and appends it to that. I've decided not to make the function create that array itself and return that, as I'm not sure if this could result in a potential double free situation (first time creating a pull request for DMD, and I'm still figuring out how DMD itself actually works).